src/contrib/cluster/Akka.Cluster.Tools/Singleton/ClusterSingletonProxy.cs

```diff
@@ -36,7 +36,7 @@ namespace Akka.Cluster.Tools.Singleton
     /// Note that this is a best effort implementation: messages can always be lost due to the distributed nature of the actors involved.
     /// </remarks>
     /// </summary>
-    public sealed class ClusterSingletonProxy : UntypedActor
+    public sealed class ClusterSingletonProxy : ReceiveActor
```

Comment:
Do you have performance comparison between ReceiveActor and Match.With? if you have, I think you should write " Contribution Guideline" and say, that we should use ReceiveActor, instead untyped actor and Match.With.

Also I'm not sure about Actor DSL, it looks like very experimental, but I saw it in some specs.